core/jobs/io/gcs_io.py

```diff
@@ -0,0 +1,199 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Provides an Apache Beam API for operating on GCS."""
+
+from __future__ import annotations
+
+from core.platform import models
+
+import apache_beam as beam
+from typing import List, Optional, Tuple, TypedDict
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import app_identity_services
+    from mypy_imports import storage_services
+
+storage_services = models.Registry.import_storage_services()
+app_identity_services = models.Registry.import_app_identity_services()
+
+BUCKET = app_identity_services.get_gcs_resource_bucket_name()
```

Comment:
Hmm, not sure if I agree with this if in the future we would for some reason want to work with different buckets we would have to refactor this I think it would be better to define the bucket in the init and `BUCKET` can be the default (maybe with different name) but there should be an option to work with different bucket.

Reply:
I agree, updated.